packages/go/analysis/ad/ad.go

```diff
@@ -482,11 +539,268 @@ func GetEdgeCompositionPath(ctx context.Context, db graph.Database, edge *graph.
+			path1CertTemplates.Add(certTemplateNode.ID.Uint32())
+			lock.Unlock()
+
+			return nil
+		})}); err != nil {
+		return nil, err
+	}
+
+	//Find all cert templates we have EnrollOnBehalfOf from our first group of templates to prefilter again
+	if err := db.ReadTransaction(ctx, func(tx graph.Transaction) error {
+		if p, err := ops.FetchPathSet(tx, tx.Relationships().Filter(
+			query.And(
+				query.InIDs(query.StartID(), cardinality.DuplexToGraphIDs(path1CertTemplates)...),
+				query.KindIn(query.Relationship(), ad.EnrollOnBehalfOf),
+				query.KindIn(query.End(), ad.CertTemplate)),
+		)); err != nil {
+			return err
+		} else {
+			enrollOnBehalfOfPaths = p
+			return nil
+		}
+	}); err != nil {
+		return nil, err
+	}
+
+	for _, path := range enrollOnBehalfOfPaths {
+		path2CandidateTemplates.Add(path.Terminal().ID.Uint32())
+	}
+
+	//Use our enterprise ca + candidate templates as filters for the third query (P2)
+	if err := traversalInst.BreadthFirst(ctx, traversal.Plan{
+		Root: startNode,
+		Driver: ADCSESC3Path2Pattern(edge.EndID, enterpriseCANodes, path2CandidateTemplates).Do(func(terminal *graph.PathSegment) error {
+			certTemplateNode := terminal.Search(func(nextSegment *graph.PathSegment) bool {
+				return nextSegment.Node.Kinds.ContainsOneOf(ad.CertTemplate)
+			})
+
+			lock.Lock()
+			path2CandidateSegments[certTemplateNode.ID] = append(path2CandidateSegments[certTemplateNode.ID], terminal)
+			path2CertTemplates.Add(certTemplateNode.ID.Uint32())
+			lock.Unlock()
+
+			return nil
+		})}); err != nil {
+		return nil, err
+	}
+
+	//EnrollOnBehalfOf is used to join P1 and P2, so we'll use it as the key
+	for _, p3 := range enrollOnBehalfOfPaths {
+		ct1 := p3.Root()
+		ct2 := p3.Terminal()
+
+		if !path1CertTemplates.Contains(ct1.ID.Uint32()) {
+			continue
+		}
+
+		if !path2CertTemplates.Contains(ct2.ID.Uint32()) {
+			continue
+		}
+
+		p1paths := path1CandidateSegments[ct1.ID]
+		p2paths := path2CandidateSegments[ct2.ID]
+
+		for _, p1 := range p1paths {
+			eca1 := p1.Search(func(nextSegment *graph.PathSegment) bool {
+				return nextSegment.Node.Kinds.ContainsOneOf(ad.EnterpriseCA) && enterpriseCANodes.Contains(nextSegment.Node.ID.Uint32())
+			})
+
+			for _, p2 := range p2paths {
+				eca2 := p2.Search(func(nextSegment *graph.PathSegment) bool {
+					return nextSegment.Node.Kinds.ContainsOneOf(ad.EnterpriseCA) && enterpriseCANodes.Contains(nextSegment.Node.ID.Uint32())
+				})
+
+				for _, p4 := range enterpriseCASegments[eca1.ID] {
+					paths.AddPath(p4.Path())
+				}
+
+				for _, p5 := range enterpriseCASegments[eca2.ID] {
+					paths.AddPath(p5.Path())
+				}
+
+				paths.AddPath(p3)
+				paths.AddPath(p1.Path())
+				paths.AddPath(p2.Path())
+
+				if collected, err := eca2.Properties.Get(ad.EnrollmentAgentRestrictionsCollected.String()).Bool(); err != nil {
+					log.Errorf("error getting enrollmentagentcollected for eca2 %d: %w", eca2.ID, err)
```

Comment:
The error wrapping here won't log well since we are trying to use the error directly so %w should be switched to %v when we are logging directly instead of returning an error up the call stack.

Reply:
Fixed

Reply:
Thank you! There are some more instances of this (I fixed a bunch in [this PR](https://github.com/SpecterOps/BloodHound/pull/292/files) that is being cherry picked into the upcoming patch). Would you mind combing through for any other instances where we are directly logging an error and using `%w` instead of `%v` please?

Reply:
Just did a find/replace to hopefully get all of them